src/components/feedback/Dialog.tsx

```diff
@@ -87,14 +85,15 @@ const Dialog = function Dialog({
   );
   const [transitionComponentVisible, setTransitionComponentVisible] =
     useState(false);
+  const isClosableDialog = !!onClose;
 
   const closeHandler = useCallback(() => {
     if (TransitionComponent) {
       // When a TransitionComponent is provided, the actual "onClose" will be
       // called by that component, once the "out" transition has finished
       setTransitionComponentVisible(false);
     } else {
-      onClose();
+      onClose?.();
```

Comment:
Technically we know that `onClose` is present at this point, but TS wouldn't, I suppose. I'm fine leaving this as-is.

Reply:
That's actually a good point 🤔 

I kind-of want to give this a thought and see if I can find a way to make TS know this is actually set at this point.